### DISCUSSION TITLE
Add vanilla-extract support

### DISCUSSION THREAD

**Discussion by cpeaustriajc** [vercel/next.js#25633] [Ideas]
### Describe the feature you'd like to request

Add native [vanilla-extract](https://github.com/seek-oss/vanilla-extract) support for next.js

### Describe the solution you'd like

Be able to use vanilla-extract without additional configurations.

I feel like this can be a good replacement for both css modules and styled-jsx because it is basically the two technologies combined.

### Describe alternatives you've considered

There is this alternative [answer](https://github.com/seek-oss/vanilla-extract/discussions/89#discussioncomment-696683) in a discussion in the repository and it manages to make it work but it throws some warnings. I have tried using it myself.

**Comment by ancashoria** [comment]
I'm in the process of installing it myself and seems to work fine during dev, but after build I don't see the required `*.vanilla.css` files and it only works at runtime (After disabling javascript, styles are not applied).

**Reply by cpeaustriajc** [reply]
Oh hey, I don't think that Next.js won't work with vanilla-extract yet because next.js has a more complex webpack configuration. I tried it also myself to configure to no avail. I always ended up having problems with it especially after the major v11 release.

Maybe I should update this feature request to make it more convincing...

**Comment by cpeaustriajc** [comment]
UPDATE: Vanilla extract now has it's own next.js plugin, I made an example repo here.

[Vanilla Extract + Next.js](https://github.com/jaycedotbin/vanilla-extract-next-js)
